stdlib/public/core/IntegerParsing.swift

```diff
@@ -146,31 +43,37 @@ extension FixedWidthInteger {
   ///     `radix`.
   ///   - radix: The radix, or base, to use for converting `text` to an integer
   ///     value. `radix` must be in the range `2...36`. The default is 10.
-  @inlinable // @specializable
-  @_semantics("optimize.sil.specialize.generic.partial.never")
+  @inlinable @inline(__always)
   public init?<S: StringProtocol>(_ text: S, radix: Int = 10) {
-    _precondition(2...36 ~= radix, "Radix not in range 2...36")
-
-    if let str = text as? String, str._guts.isFastUTF8 {
-      guard let ret = str._guts.withFastUTF8 ({ utf8 -> Self? in
-        var iter = utf8.makeIterator()
-        return _parseASCII(codeUnits: &iter, radix: Self(radix))
-      }) else {
-        return nil
-      }
-      self = ret
-      return
+    precondition(2...36 ~= radix)
+    let wholeGuts = text._wholeGuts
+    // The specialized paths require that all of the contiguous bytes can
+    // be read using UInt64 loads from (address & ~7).
+    if wholeGuts._object.isPreferredRepresentation, Self.bitWidth <= 64 {
+      let r64_ = wholeGuts.withFastUTF8({ utf8 -> UInt64? in
```

Comment:
I wonder if this would be significantly simpler to just force-contiguous the storage up front. This would incur a copy for non-small lazily bridged NSStrings. But longer term, I think this is a worthwhile tradeoff.

```swift
var copy = text
copy.withUTF8 { ... }
```

which will also handle the slicing for you (easy to forget).

Reply:
That does sounds like it might be the best approach. Would this give us the desired guarantee? Being, that the following is allowable for every code unit:
```swift
let chunk = UnsafeRawPointer(bitPattern: p & ~0x07)!.load(as: UInt64.self)
```

Reply:
Sorry, is that addressed to me? If so, could you rephrase it? (I apologize for losing track of this PR last week)

Reply:
@milseman No worries, it was but resolved it with stephen canon above (since this guarantee doesn't always hold).